src/rs.cpp

```diff
@@ -1208,6 +1275,17 @@ const char* rs2_get_option_value_description(const rs2_options* options, rs2_opt
 }
 HANDLE_EXCEPTIONS_AND_RETURN(nullptr, options, option, value)
 
+static void populate_options_list( rs2_options_list * updated_options_list,
+                                   options_watcher::options_and_values const & updated_options )
+{
+    for( auto id_value : updated_options )
+    {
+        options_watcher::option_and_value const & option_and_value = id_value.second;
+        updated_options_list->list.push_back(
+            new rs2_option_value_wrapper( id_value.first, option_and_value.p_last_known_value ) );
```

Comment:
Same. Can we use an array allocation?
Or maybe a `small_heap`?
I worry about time consumption and about memory fragmentation over time with all these periodic allocations

Reply:
If you're worried about that then we'd have a problem with every std::string usage... I don't think it's a problem.